src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
```

Comment:
Avoid using forEach:

This is the same issue mentioned in the previous comments. Instead of using a **let** declaration, followed by a **forEach**, use a **.reduce** method instead:
Same suggestion as before, sum could be solved using a reduce function
![image](https://github.com/pedro-ruas/mate-js_average-ages_test-task/assets/54222238/30f9313e-9bb9-464d-9840-f0b3ad46f7f7)